dist/init/linux-upstart/caddy.conf

```diff
@@ -1,18 +1,25 @@
-description "Caddy startup script"
+description "Caddy HTTP/2 web server"
 author "Mathias Beke"
 
 start on runlevel [2345]
 stop on runlevel [016]
 
+console logged
 
 setuid www-data
 setgid www-data
 
 respawn
 respawn limit 10 5
 
-limit nofile 4096 4096
+# Letsencrypt-issued certificates will be written to this directory.
+env HOME=/etc/ssl/caddy
+
+limit nofile 1048576 1048576
 
 script
-	exec /usr/bin/caddy -agree=true -conf=/etc/caddy/Caddyfile
-end script
\ No newline at end of file
+	mkdir -p /etc/caddy
+	mkdir -p /var/tmp/caddy
+	cd /etc/caddy
+	exec /usr/bin/caddy -agree=true -email -log=stdout -conf=/etc/caddy/Caddyfile -root=/var/tmp/caddy
```

Comment:
Incidentally (I didn't notice this before), but we can change `-agree=true` to just `-agree`. And the method of a command line flag is fine, I checked with them. Their own (former-official client) had the same thing.